### PR TITLE
[READY]Adds moonwalking

### PR DESCRIPTION
**Pull request by silicons**
https://github.com/Baystation12/Baystation12/pull/7739/files

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -977,4 +977,36 @@ var/next_mob_id = 0
...
+	if(newdir)
+		set_dir(newdir)
+		facing_dir = newdir
+	else if(facing_dir)
+		facing_dir = null
+	else
+		set_dir(dir)
+		facing_dir = dir
+
+/mob/set_dir()
```
Comment (RemieRichards) [inline code/modules/mob/mob.dm]: setDir

**Comment by kevinz000** [comment]
hold on still copying code over

**Comment by RemieRichards** [comment]
Meme.

**Comment by Thunder12345** [comment]
All these hugbox features... First picking up potted plants, now this. Nobody needs to be a pro clicker any more...

File: code/modules/mob/status_procs.dm [hunk 2]
```diff
@@ -30,20 +33,24 @@
 
 /mob/proc/Weaken(amount, updating = 1, ignore_canweaken = 0)
 	if((status_flags & CANWEAKEN) || ignore_canweaken)
+		facing_dir = null
+		facing_dir = null
```
Comment (XDTM) [inline code/modules/mob/status_procs.dm]: ?

**Comment by FantasticFwoosh** [comment]
Traitor item moonwalking boots that cause people to jump out of the way of your aggressive backwards shuffling using beepsky code when? otherwise excellent work

**Comment by InsaneHyena** [comment]
Should have added moonlighting instead

**Pull request closed without merging**